invenio_rdm_records/services/permissions.py

```diff
@@ -199,7 +201,17 @@ class RDMRecordPermissionPolicy(RecordPermissionPolicy):
         ),
     ]
     # Allow publishing a new record or changes to an existing record.
-    can_publish = can_review
+    can_publish = [
```

Comment:
Let's do the same as described above: we define a new permission `can_publish_` where, when the flag `RDM_RECORD_ALWAYS_IN_COMMUNITY` is enabled, the configured roles can still publish, and we remove the `if` from here.